dbfit-java/core/src/main/java/dbfit/fixture/ExecuteProcedureExpectException.java

```diff
@@ -4,17 +4,18 @@
 import dbfit.api.DbEnvironmentFactory;
 import dbfit.util.ExpectedBehaviour;
 import static dbfit.util.ExpectedBehaviour.*;
+import java.sql.SQLException;
 
 public class ExecuteProcedureExpectException extends ExecuteProcedure {
     private boolean excNumberDefined = false;
-    private int excNumberExpected;
+    private String excNumberExpected;
 
     public ExecuteProcedureExpectException() {
         this.environment = DbEnvironmentFactory.getDefaultEnvironment();
     }
 
     public ExecuteProcedureExpectException(DBEnvironment dbEnvironment, String procName,
-                            int expectedErrorCode) {
+                            String expectedErrorCode) {
```

Comment:
We have two potential API methods that can provide an error code/status:-
* `e.getSQLState` which returns a five character string.
* `e.getErrorCode` wihch returns the *vendor-specific* error code.

Not all database types support both methods (e.g. PostgreSQL doesn't support the `getErrorCode` method) and some database variants of SQL provide only a statement to specify an error code, to be raised, that can be recovered programmatically via only one of these methods. So what you can see in the current PR is the result of my experiments using SQL statements to specify an error code and then testing to see via which JDBC API method the code can be captured. Does that make sense?

Reply:
Yes, I can see the point. I guess in general we may have multiple variations of the fixture in future (if there is any demand for that): with the JDBC error code, JDBC sql state, and let's say `automatic` error code capturing (as in your current implementation).
Maybe the `automatic` approach is good enough for now.

Reply:
I guess we can either have separate fixtures (e.g. for SQL State and Error Code) right now (and use the current fixture for Error Code) or, as you say, have a single fixture that can trap trap both result types (the only issue being perhaps that for some future adapters it might be possible to specify either type in SQL statements).